File: app/src/main/java/com/example/techfix/ServicesActivity.java
package com.example.techfix;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.techfix.data.FirestoreCallback;
import com.example.techfix.data.RepairServiceRepository;
import com.example.techfix.model.RepairService;
import com.example.techfix.ui.ServiceAdapter;

import java.util.List;

public class ServicesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_services);

        String categoryId = getIntent().getStringExtra("CATEGORY_ID");
        String categoryName = getIntent().getStringExtra("CATEGORY_NAME");
        String userId = getIntent().getStringExtra("USER_ID");
        if (categoryName == null) categoryName = "Services";

        TextView tvTitle = findViewById(R.id.tvCategoryTitle);
        tvTitle.setText(categoryName);

        RecyclerView recyclerView = findViewById(R.id.rvServices);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        String finalUserId = userId;
        RepairServiceRepository serviceRepository = new RepairServiceRepository();
        serviceRepository.getServicesByCategory(categoryId, new FirestoreCallback<RepairService>() {
            @Override
            public void onSuccess(List<RepairService> services) {
                recyclerView.setAdapter(new ServiceAdapter(services, service -> {
                    Intent intent = new Intent(ServicesActivity.this, BookingActivity.class);
                    intent.putExtra("SERVICE_ID", service.serviceId);
                    intent.putExtra("SERVICE_NAME", service.name);
                    intent.putExtra("SERVICE_PRICE", service.price);
                    intent.putExtra("CUSTOMER_ID", finalUserId);
                    startActivity(intent);
                }));
            }

            @Override
            public void onFailure(Exception e) {}
        });
    }
}

File: app/src/main/java/com/example/techfix/ServicesActivity.kt
package com.example.techfix

import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.techfix.data.AppDatabase
import com.example.techfix.ui.ServiceAdapter
import kotlinx.coroutines.launch

class ServicesActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_services)

        val categoryId = intent.getIntExtra("CATEGORY_ID", -1)
        val categoryName = intent.getStringExtra("CATEGORY_NAME") ?: "Services"
        val userId = intent.getIntExtra("USER_ID", -1)

        findViewById<TextView>(R.id.tvCategoryTitle).text = categoryName

        val recyclerView = findViewById<RecyclerView>(R.id.rvServices)
        recyclerView.layoutManager = LinearLayoutManager(this)

        val db = AppDatabase.getDatabase(applicationContext)
        lifecycleScope.launch {
            val services = db.repairServiceDao().getServicesByCategory(categoryId)
            recyclerView.adapter = ServiceAdapter(services) { service ->
                val intent = Intent(this@ServicesActivity, BookingActivity::class.java)
                intent.putExtra("SERVICE_ID", service.serviceId)
                intent.putExtra("SERVICE_NAME", service.name)
                intent.putExtra("SERVICE_PRICE", service.price)
                intent.putExtra("CUSTOMER_ID", userId)
                startActivity(intent)
            }
        }
    }
}